src/mono/netcore/System.Private.CoreLib/src/System/Globalization/GlobalizationMode.Windows.Mono.cs

```diff
@@ -1,16 +1,39 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
+using System.Runtime.CompilerServices;
+
 
 namespace System.Globalization
 {
     internal partial class GlobalizationMode
     {
-        internal static bool UseNls => true;
-
+        internal static bool UseNls {
+            get {
+                return !Invariant && Interop.Globalization.LoadICU() == 0;
```

Comment:
Since this Property is now dependent of the `Invariant` property and both are static, they will be initialized in a static constructor in the order that they're defined. So this is dependent on the order the `GlobalizationMode` files are included.

What we did in coreclr S.P.CoreLib, to avoid depending on file compile order, we declare `Invariant` property in both the `Unix` and the `Windows` file.

Reply:
https://github.com/dotnet/runtime/blob/51fdfe19ce743c16274ebaa36f9a92b746701621/src/coreclr/src/System.Private.CoreLib/src/System/Globalization/GlobalizationMode.Windows.cs#L9-L11

https://github.com/dotnet/runtime/blob/master/src/coreclr/src/System.Private.CoreLib/src/System/Globalization/GlobalizationMode.Unix.cs#L9-L11